src/renderer/services/wallet/types.ts

```diff
@@ -33,6 +33,15 @@ export type KeystoreService = {
   validatePassword$: (password: string) => LiveData<Error, null>
 }
 
+export type HDWalletInfo = { bitcoinAddress: string }
+
+export type HDWalletState = O.Option<O.Option<HDWalletInfo>>
```

Comment:
if i understand correctly we could use here `RemoteData` type. pls check [its description](https://github.com/devexperts/remote-data-ts)